### DISCUSSION TITLE
Enable per domain statistics (feature request)

### DISCUSSION THREAD

**Discussion by Csardelacal** [postalserver/postal#1475] [Feature suggestions]
Hello Postal team! First of all, I've been running postal for a few weeks by now and I'm super happy with the app and how it works, it was a breeze to set up and a charm to work with.

I just have been having a few issues with my VPS provider being blanket blocked by some ESPs. While this is easy to detect for major providers like outlook, it's a real pain to find out whether some minor providers are blocking your server.

When I head to the outgoing tab and search for hard or soft failed messages, I get a blend of user typos and these domains that are blocking my server. Right now, I have to either look through hundreds of messages or be lucky and see a domain I am not familiar with (and then click and see whether it was indeed blocked)

I'd love to be able to have a report that tells me how many emails were sent to which domain in the last 30 days and what the failure rate has been, so I can address issues with domains that have extremely high failure rates.

**Comment by willpower232** [comment]
You could set up webhooks to report back to your app or some other software you have made to help you detect delivery problems as they occur.

Related https://github.com/atech/postal/issues/545 and https://github.com/atech/postal/issues/453

**Comment by Csardelacal** [comment]
The point I was trying to make is not about individual emails having issues being sent but entire domains blocking the mail server. When I started the server, 2/3rds of the other SMTP servers were blocking mine - and now, a month later, many of the smaller servers still do.

The point here is that when a domain (let's say outlook.com) the bounces get mixed with the regular bounces, while they're not bounces that get dealt with by a suppression but by human interaction (contacting that ESP).

I think that this is actually a problem every Mail server operator faces. And therefore is, indeed, a feature that could be a part of Postal itself.

**Comment by willpower232** [comment]
The related tickets I tagged were about reporting which is currently missing from Postal.

In the meantime, webhooks could help you report to yourself about errored emails so you don't have to go through pages and pages of messages.

**Comment by Csardelacal** [comment]
Oh! Don't worry, I don't feel like rushing you. Just wanted to point out that it'd be nice to have.
